libr/util/print.c

```diff
@@ -371,13 +371,66 @@ R_API void r_print_byte(RPrint *p, const char *fmt, int idx, ut8 ch) {
 	r_print_cursor (p, idx, 0);
 }
 
+static void print_c_code(RPrint *p, ut64 addr, ut8 *buf, int len, int ws, int w) {
+	len /= ws;
+
+	p->cb_printf ("#define _BUFFER_SIZE %d\n", len);
+
+	switch (ws) {
+	case 2:
+		p->cb_printf ("uint16_t buffer[%d] = {", len);
```

Comment:
looks like a bit spaguetti, what about passing the type as argument? or resolve the type in another helper?